### PR TITLE
(feature branch) CMR-8591: remove old-style API endpoints from ingest-app, fix generic…

### PR DESCRIPTION
**Pull request by zimzoom**
…s ingest system-int test fixture & convert it to use new API and to use generic ingest util

### CODE REVIEW

File: system-int-test/test/cmr/system_int_test/ingest/generics_test.clj [hunk 1]
```diff
@@ -103,8 +81,8 @@
    interface. Use the same native-id for all these steps"
 
   (let [native-id (format "Generic-Test-%s" (UUID/randomUUID))
-        generic-requester (partial generic-request "grid" "PROV1" native-id)
-        good-generic-requester (partial generic-requester grid-good)]
+        generic-requester (partial gen-util/generic-request-with-echo-token "grid" "PROV1" native-id)
```
Comment (eudoroolivares2016) [inline system-int-test/test/cmr/system_int_test/ingest/generics_test.clj:84]: Should this still be echo token?

**Review by eudoroolivares2016 — APPROVED** [review]

File: ingest-app/src/cmr/ingest/api/routes.clj [hunk 2]
```diff
@@ -252,13 +252,6 @@
 
 ;; TODO: Generic work - should this change to be more dynamic, gen type in front?
```
Comment (jceaser) [inline ingest-app/src/cmr/ingest/api/routes.clj:253]: comment should go right?

File: system-int-test/test/cmr/system_int_test/ingest/generics_test.clj [hunk 3]
```diff
@@ -10,40 +10,18 @@
...
-       :body (when document (json/generate-string document))
-       :throw-exceptions false
-       :headers {"Accept" "application/json"
-                 transmit-config/token-header
-                 (transmit-config/echo-system-token)}}
-      (clj-http.client/request))))
+(use-fixtures :each (join-fixtures [(ingest/reset-fixture {"provguid1" "PROV1" "provguid2" "PROV2"}
+                                                          {:grant-all-search? false
+                                                           :grant-all-ingest? false})
+                                    (dev-sys-util/freeze-resume-time-fixture)]))
```
Comment (jceaser) [inline system-int-test/test/cmr/system_int_test/ingest/generics_test.clj:24]: why are you freezing time, I don't see any tests with time

File: system-int-test/src/cmr/system_int_test/utils/generic_util.clj [hunk 4]
```diff
@@ -45,7 +45,8 @@
         :connection-manager (sys/conn-mgr)
         :body (when document (json/generate-string document))
         :throw-exceptions false
-        :headers {transmit-config/token-header token}}
+        :headers {"Accept" "application/json"
```
Comment (jceaser) [inline system-int-test/src/cmr/system_int_test/utils/generic_util.clj:48]: your version of this function is not exactly the same as mine. I can move over to yours if they were the same.

File: system-int-test/test/cmr/system_int_test/ingest/generics_test.clj (hunk 3 above)
Comment (zimzoom) [reply system-int-test/test/cmr/system_int_test/ingest/generics_test.clj:24]: you're right, i put it in mostly in anticipation of the ingest tests becoming more complex down the road, but i will remove it

File: system-int-test/test/cmr/system_int_test/ingest/generics_test.clj [hunk 5]
```diff
@@ -10,40 +10,18 @@
...
-  (-> {:method method
-       :url (url-helper/ingest-generic-crud-url concept-type provider-id native-id)
-       :connection-manager (system/conn-mgr)
-       :body (when document (json/generate-string document))
-       :throw-exceptions false
-       :headers {"Accept" "application/json"
-                 transmit-config/token-header
-                 (transmit-config/echo-system-token)}}
-      (clj-http.client/request))))
+(use-fixtures :each (join-fixtures [(ingest/reset-fixture {"provguid1" "PROV1" "provguid2" "PROV2"}
```
[the remark this answers is not in the document]
Comment (zimzoom) [reply system-int-test/test/cmr/system_int_test/ingest/generics_test.clj:21]: contacting you for more elaboration, unsure what the action step is here

File: ingest-app/src/cmr/ingest/api/routes.clj (hunk 2 above)
Comment (zimzoom) [reply ingest-app/src/cmr/ingest/api/routes.clj:253]: probably, but this wasn't my comment and so i am not 100% sure i am in a position to make that call. also, i thought that wasn't in scope for this ticket --  i assumed that we are going to be making another ticket/PR for the multitude of "clean up tasks" such as this. i am happy to remove this if someone confirms it should be removed.

File: system-int-test/test/cmr/system_int_test/ingest/generics_test.clj (hunk 1 above)
Comment (zimzoom) [reply system-int-test/test/cmr/system_int_test/ingest/generics_test.clj:84]: i am not sure, but that change is not in scope for this ticket i believe. this group of generic ingest tests was already using echo token, I just renamed it to be explicit about what it is doing. and (previously) refactored the function to be able to accept a custom token as well. 

in other words, previously:
this function used echo token, another function in providers_ingest_test used custom tokens.
now:
both groups of tests use the same function which expects a token parameter. 'generic-request-with-echo-token' is simply a partial, which defines that first parameter to be an echo token.

(see gen-util ns here, and/or pr#1673 for cmr-8380)

File: system-int-test/src/cmr/system_int_test/utils/generic_util.clj (hunk 4 above)
Comment (zimzoom) [reply system-int-test/src/cmr/system_int_test/utils/generic_util.clj:48]: I don't understand, these changes here ARE moving yours over to mine? as in, i made the needed changes in this PR here. do you mean you are doing more work independently and would like to open a PR ahead of this merge? will contact you for elaboration.

File: system-int-test/test/cmr/system_int_test/ingest/generics_test.clj (hunk 3 above)
Comment (zimzoom) [reply system-int-test/test/cmr/system_int_test/ingest/generics_test.clj]: removed

File: system-int-test/src/cmr/system_int_test/utils/generic_util.clj (hunk 4 above)
Comment (zimzoom) [reply system-int-test/src/cmr/system_int_test/utils/generic_util.clj:48]: We talked and realized that funny enough, I missed a new update that occurred while I was in the middle of rebasing this branch onto the updates that i had. 

Will rebase again on to new changes and consolidate this namespace/requester use.

File: ingest-app/src/cmr/ingest/api/routes.clj (hunk 2 above)
Comment (zimzoom) [reply ingest-app/src/cmr/ingest/api/routes.clj]: went ahead and removed this

File: system-int-test/test/cmr/system_int_test/ingest/generics_test.clj (hunk 5 above)
[the remark this answers is not in the document]
Comment (zimzoom) [reply system-int-test/test/cmr/system_int_test/ingest/generics_test.clj]: we talked, i had missed an update while rebasing, force-pushed a correct version rebased on the latest

**Review by eereiter — APPROVED** [review]

**Pull request merged**